src/lib/InstantSearch.ts

```diff
@@ -23,6 +23,7 @@ import {
   Router,
   UiState,
 } from '../types';
+import hasDetectedInsightsClient from './utils/detect-insights-client';
```

Comment:
This file could go in `lib/insights/`.

Reply:
yes both are legit places. Any good argument for having it in `lib/insights` rather than `utils/`?

Reply:
This file is related to the Insights feature. It's not a util meant to be shared and reused across the codebase. I would encourage you to read [this post about colocation](https://kentcdodds.com/blog/colocation/).

Reply:
Nice read, thank you.

It is a util, the fact that it's not used in multiple place or specialised does not make it less of a util. 
See other specialised utils it'd be colocated with: [hits-query-id](https://github.com/algolia/instantsearch.js/blob/develop/src/lib/utils/hits-query-id.ts), [geo-search](https://github.com/algolia/instantsearch.js/blob/develop/src/lib/utils/geo-search.ts#L68), [renderTemplate](https://github.com/algolia/instantsearch.js/blob/develop/src/lib/utils/renderTemplate.js#L24), [resolveSearchParameters](https://github.com/algolia/instantsearch.js/blob/develop/src/lib/utils/resolveSearchParameters.ts#L4).
One argument for putting it in `utils` is to keep  `lib/insights` stuff for public stuff (like we do for `lib/history` and `lib/stateMapping`).